src/protagonist/Engine/Ingest/Image/ImageServer/Clients/CantaloupeThumbsClient.cs

```diff
@@ -84,6 +93,37 @@ public class CantaloupeThumbsClient : IThumbsClient
         throw new HttpException(response.StatusCode, "failed to retrieve data from the thumbs processor");
     }
 
+    private HttpRequestMessage CreateCantaloupeRequestMessage(string convertedS3Location, string size)
+    {
+        var request = new HttpRequestMessage(HttpMethod.Get, $"iiif/3/{convertedS3Location}/full/{size}/0/default.jpg");
+
+        if (loadBalancerCookies.Any())
+        {
+            request.Headers.Add("Cookie", loadBalancerCookies);
+        }
+
+        return request;
+    }
+
+    private void AttemptToAddStickinessCookie(HttpResponseMessage response)
+    {
+        if (!cantaloupeClient.DefaultRequestHeaders.Contains("Cookie"))
```

Comment:
What is this check for? Is it needed?

Reply:
this check should be using the `loadBalancerCookies` list insdtead I think, and it's supposed to short circuit adding cookies to a request if they've already been added